pkg/apis/etcd/v1beta2/restore_types.go

```diff
@@ -43,8 +43,24 @@ type RestoreSpec struct {
 	// using this spec, restore operator will prepare the seed that
 	// etcd operator will pick up later.
 	ClusterSpec ClusterSpec `json:"clusterSpec"`
+
+	// BackupStorageType is the type of the backup storage which is used as RestoreSource.
+	// TODO: implement me.
+	BackupStorageType string `json:"backupStorageType"`
 	// RestoreSource tells the where to get the backup and restore from.
 	RestoreSource `json:",inline"`
+	// EtcdCluster references an EtcdCluster resource which should be failed and to be restored.
+	// TODO: implement me.
+	EtcdCluster EtcdClusterRef `json:"etcdCluster"`
+}
+
+// EtcdClusterRef tells the EtcdCluster resource which should be failed and to be restored.
```

Comment:
I don't understand what do you mean by `EtcdCluster resource which should be failed `.
isn't the point of restore operator to create a new etcdcluster with an snapshot?

Reply:
So we are assuming that the EtcdCluster resource should have "status.phase" to be "failed".

Restore operator will get the data from failed EtcdCluster and restore (delete, recreate) a new one.